pkg/controller/admissionchecks/multikueue/admissioncheck.go

```diff
@@ -54,114 +49,62 @@ func newMultiKueueStoreHelper(c client.Client) (*multiKueueStoreHelper, error) {
-
-func (a *ACReconciler) controllerFor(acName string) (*remoteController, bool) {
-	a.lock.RLock()
-	defer a.lock.RUnlock()
-
-	c, f := a.controllers[acName]
-	return c, f
-}
-
-func (a *ACReconciler) setControllerFor(acName string, c *remoteController) {
-	a.lock.Lock()
-	defer a.lock.Unlock()
-
-	if old, found := a.controllers[acName]; found {
-		old.watchCancel()
-	}
-	if c != nil {
-		a.controllers[acName] = c
-		return
-	}
-	delete(a.controllers, acName)
-}
-
-func (a *ACReconciler) Start(ctx context.Context) error {
-	a.rootContext = ctx
-	return nil
-}
 
 func (a *ACReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
 	ac := &kueue.AdmissionCheck{}
 	if err := a.client.Get(ctx, req.NamespacedName, ac); err != nil || ac.Spec.ControllerName != ControllerName {
-		if apierrors.IsNotFound(err) || !ac.DeletionTimestamp.IsZero() {
-			// stop/deleted a potential check controller
-			if cc, existing := a.controllerFor(req.Name); existing {
-				cc.watchCancel()
-				a.setControllerFor(req.Name, nil)
-				log.V(2).Info("Controller removed")
-			}
-			return reconcile.Result{}, nil
-		}
-		return reconcile.Result{}, err
+		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var remCtrl *remoteController
 	inactiveReason := ""
 
 	log.V(2).Info("Reconcile AdmissionCheck")
 	if cfg, err := a.helper.ConfigFromRef(ctx, ac.Spec.Parameters); err != nil {
 		inactiveReason = fmt.Sprintf("Cannot load the AdmissionChecks parameters: %s", err.Error())
 	} else {
-		kubeconfigs, err := a.getKubeConfigs(ctx, &cfg.Spec)
-		if err != nil {
-			a.setControllerFor(ac.Name, nil)
-			inactiveReason = fmt.Sprintf("Cannot load kubeconfigs: %s", err.Error())
-		} else {
-			remoteCtrl, existing := a.controllerFor(ac.Name)
-			if !existing {
-				remoteCtrl = newRemoteController(a.rootContext, a.client, a.wlUpdateCh)
-				if a.builderOverride != nil {
-					remoteCtrl.builderOverride = a.builderOverride
-				}
-				a.setControllerFor(ac.Name, remoteCtrl)
+		var missingClusters []string
+		var inactiveClusters []string
+		// check the status of the clusters
+		for _, clusterName := range cfg.Spec.Clusters {
+			cluster := &kueuealpha.MultiKueueCluster{}
+			err := a.client.Get(ctx, types.NamespacedName{Name: clusterName}, cluster)
+			if client.IgnoreNotFound(err) != nil {
+				log.Error(err, "reading cluster", "multiKueueCluster", clusterName)
+				return reconcile.Result{}, err
 			}
 
-			err := remoteCtrl.UpdateConfig(kubeconfigs)
 			if err != nil {
-				inactiveReason = fmt.Sprintf("Cannot start MultiKueueClusters controller: %s", err.Error())
-				a.setControllerFor(ac.Name, nil)
+				missingClusters = append(missingClusters, clusterName)
 			} else {
-				remCtrl = remoteCtrl
+				if !apimeta.IsStatusConditionTrue(cluster.Status.Conditions, kueuealpha.MultiKueueClusterActive) {
+					inactiveClusters = append(inactiveClusters, clusterName)
```

Comment:
@Maybe we can revisit this in the future, but if one cluster is inactive or even missing, we should be able to continue admitting workloads in other clusters.

Please add to the tracking list.